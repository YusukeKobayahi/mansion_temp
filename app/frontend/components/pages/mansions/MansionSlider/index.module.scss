@import "~/styles/variables";
@import "~/styles/utils";

.content {
  width: 320px;
  height: 100%;

  @include screenMd {
    width: 234px;
    margin: 0 auto;
  }
}
