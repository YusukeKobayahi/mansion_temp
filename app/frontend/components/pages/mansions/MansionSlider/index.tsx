import styles from "~/components/pages/mansions/MansionSlider/index.module.scss";
import MansionCard, {
  MansionCardProps,
} from "~/components/pages/mansions/MansionCard";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";

export interface MansionSliderProps {
  mansionCards: MansionCardProps[];
}

const MansionSlider: React.FC<MansionSliderProps> = ({
  mansionCards,
}: MansionSliderProps) => {
  const cards = mansionCards.map((o, i) => {
    return (
      <div key={i} className={styles.item}>
        <div className={styles.content}>
          <MansionCard {...o} />
        </div>
      </div>
    );
  });

  const settings = {
    dots: true,
    infinite: false,
    speed: 500,
    slidesToShow: 3,
    responsive: [
      {
        breakpoint: 768,
        settings: {
          slidesToShow: 1,
        },
      },
    ],
  };
  return (
    <div>
      <Slider {...settings}>{cards}</Slider>
    </div>
  );
};

export default MansionSlider;
