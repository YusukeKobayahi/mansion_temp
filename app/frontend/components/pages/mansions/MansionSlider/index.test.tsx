import { render } from "@testing-library/react";
import MansionSlider from "~/components/pages/mansions/MansionSlider";

describe("components/pages/mansions/MansionSlider", () => {
  it("正常に表示される", () => {
    const mansionCards = [
      {
        salesInfo: [{ info: "1000万円 ~ 1億円" }],
        href: "http://dummy.ieul.jp/ms-library/mansions/some-id",
        name: "テストマンション",
        constructedDate: "2013年05月 (築7年)",
        address: "テストアドレス",
      },
      {
        salesInfo: [{ info: "1000万円 ~ 1億円" }],
        href: "http://dummy.ieul.jp/ms-library/mansions/some-id",
        name: "テストマンション",
        constructedDate: "2013年05月 (築7年)",
        address: "テストアドレス",
      },
    ];
    const { asFragment } = render(
      <MansionSlider mansionCards={mansionCards} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
