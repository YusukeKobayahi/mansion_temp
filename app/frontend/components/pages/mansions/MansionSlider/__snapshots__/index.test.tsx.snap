// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/mansions/MansionSlider 正常に表示される 1`] = `
<DocumentFragment>
  <div>
    <div
      class="slick-slider slick-initialized"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="item"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="content"
                >
                  <div
                    class="container"
                  >
                    <button
                      class="link"
                      href="http://dummy.ieul.jp/ms-library/mansions/some-id"
                    >
                      <h3
                        class="name"
                      >
                        テストマンション
                        <br />
                        <span
                          class="date"
                        >
                          2013年05月 (築7年)
                        </span>
                      </h3>
                      <div
                        class="salesInfoArea"
                      >
                        <p>
                          販売価格
                        </p>
                        <ul>
                          <li
                            class="salesInfo"
                          >
                            1000万円 ~ 1億円
                          </li>
                        </ul>
                      </div>
                      <p
                        class="subInfo"
                      >
                        <span>
                          テストアドレス
                        </span>
                        <br />
                      </p>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="item"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="content"
                >
                  <div
                    class="container"
                  >
                    <button
                      class="link"
                      href="http://dummy.ieul.jp/ms-library/mansions/some-id"
                    >
                      <h3
                        class="name"
                      >
                        テストマンション
                        <br />
                        <span
                          class="date"
                        >
                          2013年05月 (築7年)
                        </span>
                      </h3>
                      <div
                        class="salesInfoArea"
                      >
                        <p>
                          販売価格
                        </p>
                        <ul>
                          <li
                            class="salesInfo"
                          >
                            1000万円 ~ 1億円
                          </li>
                        </ul>
                      </div>
                      <p
                        class="subInfo"
                      >
                        <span>
                          テストアドレス
                        </span>
                        <br />
                      </p>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
